ui: use status server apis for stmt bundle ops

xinhaoz opened the pull request:
Since 23.1 we've been using the sql over http api to view, request and cancel stmt bundles for fingerprints.  This caused a regression in the sql permissions required to use these features. Prior to using sql over http we only required `VIEWACTIVITY` or `VIEWACTIVITYREDACTED` for these operations. Now VIEWSYSTEMTABLE is also required since the sql over http api direclty reads from
system.statement_diagnostics_requests as the db console user.

Instead of creating a view and builtins on top of the system table, let's simply revert to the existing http apis on the status server that is already properly gated.

This change reintroduces the following APIs to DB Console
- `POST /_status/stmtdiagreports` to request statement bundles
- `GET /_status/stmtdiagreports` to list statement bundles
- `POST /_status/stmtdiagreports/cancel` to cancel statement bundles

Limitations:
This PR does not carry over the added behaviour in 23.1 where we prevent users from making multiple statement bundle reqs for a particular fingerprint. This is not really a common case in the UI since we show 'Waiting' in the overview page for the row when a stmt bundle is in progress for a fingerprint.

Epic: none
Fixes: #121301

----------------------------

ui/e2e: add e2e testing for statement bundles

This commit adds some e2e cypress tests for statement bundle features in DB Console:
* add tests for requesting, cancelling and viewing bundles for privileged (VIEWACTIVITY, ADMIN) and non-privileged users
* reformat cypress login command to use a new `users` fixture that defines users by sql privilege
* add cypress command getUserWithExactPrivileges which retrieves users from the fixture with the provided privileges

Release note: None

Part of: https://github.com/cockroachdb/cockroach/issues/121301

xinhaoz committed:
ui: use status server apis for stmt bundle ops

Since 23.1 we've been using the sql over http api to view,
request and cancel stmt bundles for fingerprints.  This
caused a regression in the sql permissions required to
use these features. Prior to using sql over http we only
required `VIEWACTIVITY` or `VIEWACTIVITYREDACTED` for these
operations. Now `VIEWSYSTEMTABLE` is also required since the
sql over http api direclty reads from
system.statement_diagnostics_requests as the db console user.

Instead of creating a view and builtins on top of the system
table, let's simply revert to the existing http apis on the
status server that are already properly gated.

This change reintroduces the following APIs to DB Console
- `POST /_status/stmtdiagreports` to request statement bundles
- `GET /_status/stmtdiagreports` to list statement bundles
- `POST /_status/stmtdiagreports/cancel` to cancel statement bundles

Limitations:
This PR does not carry over the added behaviour in 23.1 where we
prevent users from making multiple statement bundle reqs for a
particular fingerprint. This is not really a common case in the
UI since we show 'Waiting' in the overview page for the row
when a stmt bundle is in progress for a fingerprint, which prevents
another request for the same fingerprint.

Epic: none
Fixes: #121301

Release note (bug fix): Users with VIEWACTIVITY sql priv can now
request, view and cancel stmt bundles in DB console.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/128856)
<!-- Reviewable:end -->

xinhaoz committed:
ui/e2e: add e2e testing for statement bundles

This commit adds some e2e cypress tests for statement
bundle features in DB Console:
* add tests for requesting, cancelling and viewing bundles
for privileged (VIEWACTIVITY, ADMIN) and non-privileged
users
* reformat cypress login command to use a new `users`
fixture that defines users by sql privilege
* add cypress command getUserWithExactPrivileges which
retrieves users from the fixture with the provided
privileges

Release note: None

Part of: #121301

dhartunian reviewed (commented):
Do we also need to provide the ability to download the bundle? I get an error when I try to do that with `VIEWACTIVITY` because this endpoint needs to be updated

https://github.com/cockroachdb/cockroach/blob/99d10d419c08d68db4e3a5d62c9dec009d6cdef6/pkg/server/admin.go#L265-L287

> Reviewed 1 of 1 files at r1, 1 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/128856#-:-O4qU58E0WTZTjEoqJ_H:b-pp8n8f)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @angles-n-daemons and @xinhaoz)

___
*[`pkg/ui/workspaces/e2e-tests/cypress/e2e/statementBundles/statementBundles.cy.ts` line 85 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/128856#-O4qPoXyAeMUEzXvhQoR:-O4qPoXz3X1Sf041Y9eI:bnvn1ye) ([raw file](https://github.com/cockroachdb/cockroach/blob/9e858d0727e3b16799cda5524a670950eb3d4232/pkg/ui/workspaces/e2e-tests/cypress/e2e/statementBundles/statementBundles.cy.ts#L85)):*
> ```XML
> 
>   describe("non-privileged VIEWACTIVITYREDACTED user", () => {
>     runTestsForNonPrivilegedUser(SQLPrivilege.VIEWACTIVITY);
> ```

this should be `VIEWACTIVITYREDACTED`?


<!-- Sent from Reviewable.io -->

dhartunian reviewed (approved):
oh duh it's the next PR :facepalm: please ignore. this LGTM with just the nit in the e2e test.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/128856#-:-O4qUXw9CzyGUdo8S_xT:b-u9hx6b)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @angles-n-daemons and @xinhaoz)



<!-- Sent from Reviewable.io -->

xinhaoz commented:
TFTR! Addressed nit.
bors r+
